zerver/webhooks/splunk/view.py

```diff
@@ -0,0 +1,37 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Iterable, Optional, Text
+
+@api_key_only_webhook_view('Splunk')
+@has_request_variables
+def api_splunk_webhook(request, user_profile, client,
+                       payload=REQ(argument_type='body'), stream=REQ(default='splunk'),
+                       topic=REQ(default='Splunk Alert')):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], Text, Optional[Text]) -> HttpResponse
+
+    # use default values if expected data is not provided
+    search_name = payload.get('search_name', 'Missing search_name')
+    results_link = payload.get('results_link', 'Missing results_link')
+    host = payload.get('result', {}).get('host', 'Missing host')
+    source = payload.get('result', {}).get('source', 'Missing source')
+    raw = payload.get('result', {}).get('_raw', 'Missing _raw')
```

Comment:
The only thing that can send an outgoing webhook (that I can find) is a saved search. You can configure an action to perform when that search is executed, and one of them is a webhook notification. Each individual search can have its own. 

For programmatically creating a topic, the name of the search would make the most sense. Names can be up to 100 characters, is that too long?